src/main/java/de/hhu/propra/link/services/LinkService.java

```diff
@@ -17,45 +20,83 @@ public LinkService(LinkRepository linkRepository) {
     /**
      * Creates a not yet used viable abbreviation for the entered URL.
      *
-     * @param url The URL that is to be shortened
+     * @param urlString The URL that is to be shortened
      * @return the abbreviation for the long URL
      */
-    private String autoAbbreviation(String url) {
-        String schemelessUrl = removeProtocol(url);
-        if (schemelessUrl.indexOf("www") == 0) {
-            schemelessUrl = schemelessUrl.substring(3);
+    private String autoAbbreviation(String urlString) {
+        URL url;
+        try {
+            url = new URL(urlString);
+        } catch (MalformedURLException e) {
+            throw new IllegalArgumentException(e);
```

Comment:
Who is catching this Exception? Throwing an exception to the user is not a good idea. There should be an error message in the frontend instead.

Reply:
Spring was catching it and then showing it, which is no good. Will handle more gracefully.